src/Schema/AST/ASTHelper.php

```diff
@@ -195,11 +188,17 @@ public static function hasDirective(Node $definitionNode, string $name): bool
     /**
      * Out of a list of nodes, get the first that matches the given name.
      *
-     * @param  iterable<\GraphQL\Language\AST\Node> $nodes
+     * @template TNode of \GraphQL\Language\AST\Node
+     * @param  iterable<TNode> $nodes
+     * @return TNode|null
      */
     public static function firstByName($nodes, string $name): ?Node
     {
         foreach ($nodes as $node) {
+            if (! property_exists($node, 'name')) {
+                throw new Exception('Expected a Node with a name property, got: '.get_class($node));
```

Comment:
Should this be another type of exception? 👍 
Just thinking if this should be a custom exception or a RuntimeException?

Reply:
Definitely should be, I would probably do an overhaul of all the Exceptions in one go. I like to cut through such concerns vertically in order to devise a wholistic and consistent solution for the whole library.